### ISSUE TITLE
feature request: open file in order of Spaced repetition techniques

### ISSUE THREAD

**Issue by ryjjin** [erichalldev/obsidian-smart-random-note#7]
Hi Eric! Great plugin, I have been happy then seen it.

Is it possible to make the chance of opening a random note not equally probable (or equally probable, except for the already viewed notes), but take into account the [spaced repetition techniques](https://en.wikipedia.org/wiki/Spaced_repetition)?

The idea is to give the highest priority to notes written (or edited, if possible) in the last 24 hours, and then gradually reduce the likelihood of opening for notes written over several days - a week - several weeks, and so on.

If, at the same time, add the condition that the previously viewed notes are not opened by the plugin (an idea that you wrote about in your plans for the future in the plugin description) - then here you will additionally need to reset the "viewed" status of the note when it goes to another priority group

**Comment by erichalldev** [comment]
Hey @ryjjin, thanks for the suggestion! I've mulled over adding spaced repetition capabilities and I think I'd like to implement it. To be clear, I don't want to make a flash card system like Anki, rather a simple system to help give visibility to your whole vault.

I don't know when I'll be able to get to it but I'll put it on the roadmap. I want to make sure I produce an intuitive, well thought-out implementation.

**Comment by ryjjin** [comment]
Thanks! 
Yes, I don't see it as a flash card system, but rather as a way to prioritize random notes based on the age of the note.  

And I just realized that I wrote above about the fact that it may be worth starting from the date the file was modified, but it would probably be more correct to look after the creation date (most people will probably use this more often in order to possibly rethink and add past notes,  rather than trying to remember the changes made)

**Comment by base-I** [comment]
> Hey @ryjjin, thanks for the suggestion! I've mulled over adding spaced repetition capabilities and I think I'd like to implement it. To be clear, I don't want to make a flash card system like Anki, rather a simple system to help give visibility to your whole vault.
> 
> I don't know when I'll be able to get to it but I'll put it on the roadmap. I want to make sure I produce an intuitive, well thought-out implementation.

Exactly what I am looking! Excited to see this implemented in the future :)

**Comment by base-I** [comment]
I thought about this for a while now: For my purposes it should be enough to show me the note from a search I have not seen in the longest. It is just for rough review of topics; should I want a "precise" spaced repitition, I'd make an anki flashcard

**Comment by erichalldev** [comment]
@ryjjin and @base-I, I've decided that I want to shift my focus away from implementing spaced repetition since other plugins have done this since we discussed it. See the updated Readme for those plugins. Do you think those could serve your needs?

Additionally, even showing the note that hasn't been seen for the longest time isn't random. I'd like to stay as close as possible to opening random notes with greater control. 

It might be test to close this issue. What are your thoughts?

**Comment by base-I** [comment]
Agree, I think a mixture of the incremental writing plugin and the Anki integration does the trick for me :)
